src/tests/ngraph_helpers/snippets_ngraph_functions/include/two_binary_ops_function.hpp

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/ngraph.hpp>
+#include "ngraph/opsets/opset1.hpp"
+#include "snippets_helpers.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+/*
+ * @class BaseDummyOperation
+ * @brief Base not type relaxed operation with absent validation and parameterized output.
+ */
+class BaseDummyOperation : public ngraph::op::Op {
+public:
+    BaseDummyOperation(
+        const Output<Node>& arg0,
+        const Output<Node>& arg1,
+        const element::Type& output_type = element::undefined) : Op({ arg0, arg1 }), output_type(output_type) {
+        constructor_validate_and_infer_types();
+    }
+
+    void validate_and_infer_types() override {
+        set_output_type(
+            0,
+            output_type == element::undefined ? get_input_element_type(0) : output_type,
+            get_input_partial_shape(0));
+    }
+
+    element::Type get_output_type() const { return output_type; }
+
+protected:
+    element::Type output_type;
+};
+
+/*
+ * @class DummyOperation1
+ * @brief Not type relaxed operation with absent validation and parameterized output.
+ */
+class DummyOperation1 : public BaseDummyOperation {
+public:
+    OPENVINO_OP("DummyOperation1", "test::snippets");
+
+    DummyOperation1(
+        const Output<Node>& arg0,
+        const Output<Node>& arg1,
+        const element::Type& output_type = element::undefined) : BaseDummyOperation(arg0, arg1, output_type) {
+        constructor_validate_and_infer_types();
+    }
+
+    DummyOperation1(const BaseDummyOperation& op) : DummyOperation1(op.get_input_source_output(0), op.get_input_source_output(1), op.get_output_type()) {
+        constructor_validate_and_infer_types();
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<DummyOperation1>(new_args.at(0), new_args.at(1), this->get_output_type());
+    }
+};
+
+/*
+ * @class DummyOperation2
+ * @brief Not type relaxed operation with absent validation and parameterized output.
+ */
+class DummyOperation2 : public BaseDummyOperation {
+public:
+    OPENVINO_OP("DummyOperation2", "test::snippets");
+
+    DummyOperation2(
+        const Output<Node>& arg0,
+        const Output<Node>& arg1,
+        const element::Type& output_type = element::undefined) : BaseDummyOperation(arg0, arg1, output_type) {
+        constructor_validate_and_infer_types();
+    }
+
+    DummyOperation2(const BaseDummyOperation& op) : DummyOperation2(op.get_input_source_output(0), op.get_input_source_output(1), op.get_output_type()) {
+        constructor_validate_and_infer_types();
+    }
+
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
+        return std::make_shared<DummyOperation2>(new_args.at(0), new_args.at(1), this->get_output_type());
+    }
+};
+
+/*
+ * @class TwoElementWiseFunction
```

Comment:
Please update the description to match the class name.

Reply:
done, thanks